packages/core/database/lib/query/query-builder.js

```diff
@@ -472,10 +473,18 @@ const createQueryBuilder = (uid, db, initialState = {}) => {
       try {
         const qb = this.getKnexQuery();
 
+        if (transactionCtx.get()) {
+          qb.transacting(transactionCtx.get());
+        }
```

Comment:
Good question if you do transacting inside an already running transaction it will use the current one. We probably can support nested transactions if there is a usecase for it

Reply:
One of the deadlock fixes consisted on making two queries not on the same transaction (problem was using MySQL and MariaDB)
https://github.com/strapi/strapi/pull/15034/files#diff-1b25302e2d8e3199d9b2b9c2b238caa65f995026e545fafae5a2cb5a516a08bdR300

So maybe if they do share the same transaction with this change we would be reintroducing the same issue? I will play with this.

Reply:
Probably yes. Let me know if you reproduce it

Reply:
@Marc-Roig did you manage to test this ?
i just did some changes that i guess might introduce the deadlock issue again, I would like to have some context if possible

Reply:
Not yet unfortunately, I have not been able to reproduce the MySQL deadlock issues :( what db are you using?